runner.go

```diff
@@ -84,26 +88,46 @@ func (r *Runner) Run() {
 	}
 
 	for {
-		if err := r.runOnce(); err != nil {
+
+		if r.rebuildSchedule != nil {
+			//if there is a pending rebuild catch it otherwise do normal thing.
+			select {
+			case <-r.rebuildSchedule.C:
+				log.Logf("running scheduled rebuild (%s)", time.Now().Format(time.RFC3339))
+				if err := r.runOnce(true); err != nil {
+					log.Logf("runner encountered error: %s", err.Error())
+				}
+				continue
```

Comment:
Should it sleep `runInterval` and do the backoff before continuing to the next iteration? Maybe we could just set a variable in the `select` block and pass it to the `r.runOnce` call bellow.

Reply:
I think i'll change it so that on the first run it will always skip the sleep interval but ordinarily it will take the rebuild into account and back-off.